test-project/Assets/EditmodeTests/Collections/ReferenceProviderTests.cs

```diff
@@ -35,25 +29,30 @@ public void Disposed_world_cleans_provider()
         [Test]
         public void Removed_component_disposes_reference()
         {
+            var template = GetTemplate();
             using (var mockWorld = MockWorld.Create(new MockWorld.Options()))
             {
-                mockWorld.Step(world =>
-                {
-                    world.Connection.CreateEntity(entityId, new EntityTemplate());
-
-                    world.Connection.AddComponent(entityId, Position.ComponentId,
-                        new Position.Update {Coords = Coordinates.Zero});
-
-                    world.Connection.AddComponent(entityId, Metadata.ComponentId,
-                        new Metadata.Update {EntityType = "EntityWithReferenceComponent"});
-                }).Step(world =>
-                {
-                    Assert.AreEqual(1, ReferenceProvider<string>.Count);
-                    world.Connection.RemoveComponent(entityId, Metadata.ComponentId);
-                });
+                mockWorld
+                    .Step(world =>
+                    {
+                        world.Connection.CreateEntity(entityId, template);
+                    })
+                    .Step(world =>
+                    {
+                        Assert.AreEqual(1, ReferenceProvider<string>.Count);
+                        world.Connection.RemoveEntityAndComponents(entityId, template);
+                    });
 
                 Assert.AreEqual(0, ReferenceProvider<string>.Count);
             }
         }
+
+        private EntityTemplate GetTemplate()
+        {
+            var template = new EntityTemplate();
+            template.AddComponent(new Position.Snapshot(), "some-worker");
+            template.AddComponent(new Metadata.Snapshot(), "some-worker");
```

Comment:
This should have a value to keep the tests the same

Reply:
Hahaha this actually means we pushed a `null` into the provider